torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -0,0 +1,366 @@
+#include <torch/csrc/jit/codegen/cuda/lower_expr_sort.h>
+
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_utils.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+namespace {
+
+TensorView* findOutputTensor(Expr* expr) {
+  TORCH_INTERNAL_ASSERT(
+      expr->outputs().size() <= 1, "Unexpected number of outputs");
+  if (expr->outputs().size() != 1) {
+    return nullptr;
+  }
+  auto out = expr->output(0);
+  if (out->getValType() != ValType::TensorView) {
+    return nullptr;
+  }
+  return out->as<TensorView>();
+}
+
+void findTargetTensor(Expr* expr, TensorView*& target, unsigned& score) {
```

Comment:
why not return score instead of the out param?

Reply:
`target` is also an out parameter. Could change it to return a pair.

Reply:
no std::pair please - but a dedicated struct would look nicely.

Reply:
That'd seem a little too much to me. This is just a helper function only used in this source file.

Reply:
why is it too much? a struct would accomplish more than a comment, with about the same about of lines. And we'd also improve the overall code structure by allowing a functional style instead of out parameters.

anyway, this is a suggestion, feel free to leave the code as is if you feel strongly about it

Reply:
Should out parameters not be used? How should, say, in-place modification of vectors, be written? In-place isn't functional, but is it more preferred despite its performance concern?

Reply:
what is the performance concern? 

out parameters are problematic for many reasons, especially in languages where there's no distinction between out vs in/out parameters, like in C++. We can probably find a few cases where out parameters might make sense, but in general I try to stay away from them. And if you absolutely must use out parameters, use pointers, which makes it more explicit that you're not passing by value.

Reply:
Let's say just want to modify a vector by a function. Passing it as a reference to the function and doing in-place modification in the function would be what I typically do. How would you do this without using out parameters?

Reply:
in-place update of a very large object would be a place where you could use *in/out* parameters. but that's not the case here (we don't have large objects and we don't have in/out, just out). have I misunderstood the question?

Reply:
No, that's not the case in this particular code. I was just wondering what to do without (in)/out parameters. 

I updated the code to return a struct.